src/components/inputs/commons/InputLabel.tsx

```diff
@@ -4,24 +4,20 @@
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
-import styled, { SimpleInterpolation } from 'styled-components';
+import styled from 'styled-components';
 
 import { getColor } from '../../../theme/theme-utils';
 
-export const InputLabel = styled.label.attrs<
-	{
-		$hasError?: boolean;
-		$hasFocus?: boolean;
-		$disabled?: boolean;
-	},
-	{ $textColor: string }
->(({ $hasError, $hasFocus }) => ({
-	$textColor: ($hasError && 'error') || ($hasFocus && 'primary') || 'secondary'
-}))<{
+type InputLabelProps = {
 	$hasError?: boolean;
 	$hasFocus?: boolean;
 	$disabled?: boolean;
-}>`
+	$textColor?: string;
+};
+
+export const InputLabel = styled.label.attrs<InputLabelProps>(({ $hasError, $hasFocus }) => ({
```

Comment:
This refactor is wrong. Attrs function is used to create a new attribute, not settable from outside, which is composed starting from the other props.
We need to understand how the attrs types changed and update it accordingly, but placing textColor in the exposed attributes is not right.